### PR TITLE
[Console] Log section

### PR DESCRIPTION
**Pull request by avanderhoorn**
This adds log section which displays browser and server console log Error, Warn and Info details:
jun-25-2017 19-59-24

### CODE REVIEW

File: src/proxy/console.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const methods = [
+    'error',
+    'info',
+    'warn'
+];
+const listeners = [];
+
+methods.forEach(function(methodKey) {
```
Comment (davidkpiano) [inline src/proxy/console.js:10]: I'd add a todo comment here to get this directly from `Glimpse.Browser.Agent` in the future.

File: src/proxy/console.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+            return oldFunction.apply(this, arguments);
+        };
+
+        return newFunction;
+    }(methodKey));
+});
+
+function publishLog(details) {
+    for (let i = 0; i < listeners.length; i++) {
```
Comment (davidkpiano) [inline src/proxy/console.js]: Super tiny nit: `listeners.forEach(listener => listener(details))` (for sake of comprehension/readability)

File: src/views/logs.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
+const dom = require('../lib/dom');
+const consoleProxy = require('../proxy/console');
+const summaryRepository = require('../repository/summary');
+
+const supporedLevels = [ 'error', 'warn', 'info' ];
```
Comment (davidkpiano) [inline src/views/logs.js]: Typo: `supportedLevels` (t after `suppor`)

File: src/views/logs.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                </div>
+            </div>
+        `;
+    },
+    postRender: function() {
+        state.ready = true;
+
+        // run through the browser logs if we have any
+        state.preRenderCache.forEach(function(task) {
+            task();
```
Comment (davidkpiano) [inline src/views/logs.js]: Tiny nit: name this `doTask()`? It's a side-effectful (imperative) function but `task()` doesn't indicate that.

**Comment by davidkpiano** [comment]
Code LGTM, a couple nits.

**Review by davidkpiano — APPROVED** [review]

File: src/proxy/console.js (hunk 1 above)
Comment (avanderhoorn) [reply src/proxy/console.js:10]: Added.

File: src/proxy/console.js (hunk 2 above)
Comment (avanderhoorn) [reply src/proxy/console.js]: Fixed.

File: src/views/logs.js (hunk 3 above)
Comment (avanderhoorn) [reply src/views/logs.js]: Fixed.

File: src/views/logs.js (hunk 4 above)
Comment (avanderhoorn) [reply src/views/logs.js]: Adjusted.

**Comment by nikmd23** [comment]
Can we remove the `/` delimiters and instead put the proper icon in front of each number.

I don't know if that's Info / Warning / Error, or Error / Warning / Info - and I on;y know that it's one of those three values because I defined the requirements.

This format would be much better:

❗️ X ⚠️ Y  ℹ️ Z

**Comment by nikmd23** [comment]
Also, lets move the logs section to the left, before Server calls and after Load time.

That way, the two sections on the right (Ajax and Open in) are about navigation and not interrupted.

**Pull request merged**